nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -442,10 +437,10 @@ in
         ";
       };
 
-      stateDir = mkOption {
-        default = "/var/spool/nginx";
+      stateDirName = mkOption {
```

Comment:
Is hard coding the state directory to `/var/lib/nginx` a bad idea?

Reply:
Systemd only gives me the possibility to name a directory under /usr/lib as state directory. Absolute paths and .. don't work. Using another directory means I'd have to take care about mkdir, chown and chmod myself. This way, systemd does all that for me.
The only tradeoff is that the directory is moved from /var/spool, but /var/lib is the more appropriate location imo. Maybe we can get rid of the logs directory as NixOS's nginx logs errors to stderr and has no access log enabled by default.

Reply:
Yes I'm all for locking the state directory down under `/var/lib/` my question was asking if the directory name under `/var/lib/` needs to be an option? Could the state directory not be hard coded to `/var/lib/nginx` (in other words `StateDirectory = "nginx";`? Is there value to the user in allowing choice of the directory name here?

Reply:
It is possible to leave use custom folder?